deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -85,11 +85,52 @@ export class StackComposer {
         } else if (engineVersionString && engineVersionString.startsWith("ES_")) {
             version = EngineVersion.elasticsearch(engineVersionString.substring(3))
         } else {
-            throw new Error("Engine version is not present or does not match the expected format, i.e. OS_1.3 or ES_7.9")
+            throw new Error(`Engine version (${engineVersionString}) is not present or does not match the expected format, i.e. OS_1.3 or ES_7.9`)
         }
         return version
     }
 
+    private getBasicClusterAuth(basicAuthObject: { [key: string]: any }): ClusterBasicAuth {
```

Comment:
nit: Could pull these auth object parsing methods to their own class to reduce the already high complexity of this file